app/src/main/java/de/ur/explure/views/ProfileFragment.kt

```diff
@@ -1,6 +1,236 @@
 package de.ur.explure.views
 
+import android.app.Activity.RESULT_OK
+import android.app.AlertDialog
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.net.Uri
+import android.os.Bundle
+import android.provider.MediaStore
+import android.provider.MediaStore.ACTION_IMAGE_CAPTURE
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
 import androidx.fragment.app.Fragment
+import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
+import com.crazylegend.viewbinding.viewBinding
+import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.textfield.TextInputLayout
+import com.google.firebase.storage.FirebaseStorage
+import de.ur.explure.GlideApp
 import de.ur.explure.R
+import de.ur.explure.databinding.FragmentProfileBinding
+import de.ur.explure.extensions.toDp
+import de.ur.explure.viewmodel.ProfileViewModel
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.component.inject
+import java.io.ByteArrayOutputStream
 
-class ProfileFragment : Fragment(R.layout.fragment_profile)
+class ProfileFragment : Fragment(R.layout.fragment_profile) {
+
+    private val binding by viewBinding(FragmentProfileBinding::bind)
+
+    private val viewModel: ProfileViewModel by viewModel()
+
+    private val fireStorage: FirebaseStorage by inject()
+
+    private var userId: String = ""
+    private lateinit var imageUri: Uri
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setHasOptionsMenu(true)
+
+        setOnClickListeners()
+
+        observeUserModel()
+        viewModel.getUserInfo()
+    }
+
+    private fun observeUserModel() {
+        viewModel.user.observe(viewLifecycleOwner, { user ->
+            if (user != null) {
+                binding.userNameTextView.text = user.name
+                userId = user.id
+                if (user.profilePictureUrl.isNotEmpty()) {
+                    try {
+                        val gsReference =
+                            fireStorage.getReferenceFromUrl(user.profilePictureUrl)
+                        GlideApp.with(requireContext())
+                            .load(gsReference)
+                            .error(R.drawable.user_profile_picture)
+                            .transition(DrawableTransitionOptions.withCrossFade())
+                            .into(binding.profilePicture)
+                    } catch (_: Exception) {
+                    }
+                }
+            }
+        })
+    }
```

Comment:
Also der Code um das Profilbild zu updaten ist in jedem Fragment im Profil, das ist etwas redundant, aber auch nicht allzu schlimm.